tests/test_app.py

```diff
@@ -288,6 +288,64 @@ def test_get_title_returns_page_containing_official_copy_info_page_when_present(
                 assert response_body.index(string_to_find) > response_body.index(previous_string)
 
 
+class TestDisplayTitlePdf(BaseServerTest):
+
+    def setup_method(self, method):
+        self.app = app.test_client()
+        self._log_in_user()
+
+        with mock.patch('service.login_api_client.authenticate_user', return_value=True):
+            self._log_in_user()
+
+    @mock.patch('service.api_client.requests.get', return_value=unavailable_title)
+    def test_display_title_pdf_no_title(self, mock_get):
+        response = self.app.get('/titles/titleref.pdf')
+        assert response.status_code == 404
+        assert 'Page not found' in response.data.decode()
+
+    @mock.patch('service.api_client.requests.get', return_value=fake_title)
+    @mock.patch('service.api_client.get_official_copy_data', return_value=official_copy_response)
+    def test_display_title_pdf(self, mock_get_official_copy_data, mock_get):
+        response = self.app.get('/titles/titleref.pdf')
+        assert response.status_code == 200
+
+    @mock.patch('service.api_client.requests.get', return_value=fake_title)
+    @mock.patch('service.api_client.get_official_copy_data', return_value=official_copy_response)
+    @mock.patch('service.server.render_template')
+    def test_display_title_pdf_renders_template(self, mock_render, mock_get_official_copy_data,
+                                                mock_get):
+        self.app.get('/titles/titleref.pdf')
+        actual_call = mock_render.mock_calls[0]
+        actual_args = actual_call[1]
+        actual_kwargs = actual_call[2]
+
+        assert actual_args[0] == 'full_title.html'
+        assert actual_kwargs['title_number'] == 'titleref'
+
+        import datetime  # TODO: use assert_called_once_with once we get publication_date from API
+        assert type(actual_kwargs['publication_date']) == datetime.datetime
```

Comment:
How about comparing this to the actual date so that when we change it in the code, this test will start breaking and we'll change it properly?